### PR TITLE
Allow unknown in 'no-object-literal-type-assertion' rule.

### LINKED ISSUE #4355: no-object-literal-type-assertion should allow unknown type
### Feature request

**Is your feature request related to a problem? Please describe.**
There is 'as any' exception for the rule. #2716
With the new unknown type there should be exception for 'as unknown' as well.

**Describe the solution you'd like**
Add exception for 'as unknown'.

### PR DESCRIPTION
**Pull request by NN---**
Fix #4355

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/tslint](https://github.com/palantir/tslint), @NN---! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: src/rules/noObjectLiteralTypeAssertionRule.ts [hunk 1]
```diff
@@ -58,6 +58,10 @@ function walk(ctx: Lint.WalkContext<void>): void {
         if (
             isAssertionExpression(node) &&
             node.type.kind !== ts.SyntaxKind.AnyKeyword &&
+            // Compare with UnknownKeyword if using TS 3.0 or above
+            (!!(ts.SyntaxKind as any).UnknownKeyword ?
+             (node.type.kind !== (ts.SyntaxKind as any).UnknownKeyword) :
+             true) &&
```
Comment (JoshuaKGoldberg) [inline src/rules/noObjectLiteralTypeAssertionRule.ts]: @NN--- if you change the `true` to `node.type.getText(ctx.sourceFile) !== "unknown"` the `<unknown> {};` line will be properly linted, so you can remove the expected error from it.

Comment (NN---) [reply src/rules/noObjectLiteralTypeAssertionRule.ts]: Just to make sure, in TS 2 there is no unknown keyword and this rule can be abused.
I think these cases are rare.

```ts
type unknown = {};
{} as unknown // Not what we wanted.
```

Comment (NN---) [reply src/rules/noObjectLiteralTypeAssertionRule.ts]: My point is that you could make anything in TS 2.0 as unknown is not a keyword.
Perhaps not a clever idea but still :)

Comment (JoshuaKGoldberg) [reply src/rules/noObjectLiteralTypeAssertionRule.ts]: Yes understood - and I think that's ok for this rule because, as you noted, it's not a clever idea.

**Pull request merged**